### DISCUSSION TITLE
Does cpptools support preprocessor like #ifdef ?

### DISCUSSION THREAD

**Discussion by daxianji007** [microsoft/vscode-cpptools#9038] [Q&A] [answered]
For example in the screenshot, there should be only one struct. But the intellisense shows there are two, and Struts2 is recognized as a child of Struct1. That is weird.
![image](https://user-images.githubusercontent.com/8327019/158728481-1d00f491-211d-4c84-b8a1-8c3b6c3cb7c6.png)

Does cpptools support preprocessor like #ifdef ? Or is there any way to workaround this?

**Comment by sean-mcmanus** [comment, marked as answer]
That's a side of effect of our tag parser mode (global symbol parsing) which doesn't evaluate macro defines. Somewhat "by design". I don't know of a workaround.  You could try filing an issue at https://developercommunity.visualstudio.com/ and see what they say.
